common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -314,6 +322,67 @@ public static String simpleClassName(Class<?> clazz) {
         }
     }
 
+    /**
+     * Escapes the specified value, if necessary according to
+     * <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>.
+     * <p>
+     * The {@link CharSequence} will be escaped is if contains a
+     * <ul>
+     *     <li>{@link StringUtil#COMMA}</li>
+     *     <li>{@link StringUtil#LINE_FEED}</li>
+     *     <li>{@link StringUtil#CARRIAGE_RETURN}</li>
+     *     <li>{@link StringUtil#DOUBLE_QUOTE}</li>
+     * </ul>
+     * If a double quote is found inside of the value, it will be escaped by preceeding it
+     * with another double quote.
+     *
+     * @param value the value to be appended, and quoted if necessary
+     * @return {@link CharSequence} the quoted value
+     */
+    public static CharSequence escapeCsv(CharSequence value) {
+        int length = checkNotNull(value, "value").length();
+        if (length == 0) {
+            return value;
+        }
+        int last = length - 1;
+        boolean quoted = isDoubleQuote(value.charAt(0)) && isDoubleQuote(value.charAt(last)) && length != 1;
+        boolean foundSpecialCharacter = false;
+        boolean foundEscapedCharacter = false;
+        StringBuilder escaped = new StringBuilder(length).append(DOUBLE_QUOTE_STRING);
+        for (int i = 0; i < length; i++) {
+            char current = value.charAt(i);
+            switch (current) {
+                case DOUBLE_QUOTE:
+                    if ((i == 0 || i == last) && quoted) {
+                        continue;
+                    }
+                    // last charater but value is not quoted, so it needs to be escaped.
+                    if (i == last) {
```

Comment:
To reduce the conditional blocks we could roll the `if` check above (i.e. `if ((i == 0 || i == last) && quoted)`) into this if/else block.  WDTY?

```java
if (i == last) {
  if (quoted) continue;
  escaped.append(DOUBLE_QUOTE);
} else if (i == 0 && quoted) {
  continue;
} else {
...
}
```